lib/spack/spack/hooks/autopush.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import argparse
+
+import spack.mirror
+from spack.cmd import buildcache as bc
+
+
+def post_install_in_db(spec):
+    pkg = spec.package
+    # Package was installed from source
+    if not pkg.installed_from_binary_cache:
+        hash = "/" + spec.dag_hash()
+        # Iterate over all mirrors with autopush==True
+        all_mirrors = spack.mirror.MirrorCollection(binary=True)
+        autopush_mirrors = [mirror for mirror in all_mirrors.values() if mirror.get_autopush()]
+        for mirror in autopush_mirrors:
+            # Create and execute buildcache push command
+            bc_parser = argparse.ArgumentParser()
+            bc.setup_parser(bc_parser)
+            bc_flags = []
+            # The dependencies should already be in the buildcache
+            # No need to add them again (especially with --force)
+            bc_flags.append("--only=package")
+            # We want the pushed packages to be available right away
+            bc_flags.append("--update-index")
```

Comment:
I don't think `--update-index` should be a hard-coded default. `--update-index` is a linear time operation so if you do it for every push it becomes quadratic.

Reply:
I moved the update of the index until the end of installation, so it is done only once. I created a new hook for that.
What do you think? Do you have an idea how to call that hook?